solidity/src/proof_gadgets/MembershipCheck.pre.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: UNLICENSED
+// This is licensed under the Cryptographic Open Software License 1.0
+pragma solidity ^0.8.28;
+
+import "../base/Constants.sol";
+import "../base/Errors.sol";
+import {VerificationBuilder} from "../builder/VerificationBuilder.pre.sol";
+
+/// @title MembershipCheck
+/// @dev Library for handling membership checks
+library MembershipCheck {
+    /// @notice verifies that the first n rows of a column contain the first m rows of another column
+    /// @custom:as-yul-wrapper
+    /// #### Wrapped Yul Function
+    /// ##### Signature
+    /// ```yul
+    /// membership_check_evaluate(builder_ptr, alpha, beta, chi_n_eval, chi_m_eval, column_evals, candidate_evals)
+    /// ```
+    /// ##### Parameters
+    /// * `builder_ptr` - memory pointer to the verification builder
+    /// * `alpha` - a challenge
+    /// * `beta` - a challenge
+    /// * `chi_n_eval` - the evaluation of a column of 1s with length n, where n is the number of rows that should contain the relevant candidate rows
+    /// * `chi_m_eval` - the evaluation of a column of 1s with length m, where m is the number of rows that should be contained by the relevant column rows
```

Comment:
I think n and m are flipped in the comments compared to what they are in Rust and Solidity code